File: proj.android_studio/ee-x-facebook-ads/src/main/java/com/ee/facebook/FacebookBannerHelper.java
package com.ee.facebook;

import android.content.res.Resources;
import android.graphics.Point;

import androidx.annotation.NonNull;

import com.ee.core.internal.Utils;
import com.facebook.ads.AdSize;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

import static com.google.common.truth.Truth.assertThat;

class FacebookBannerHelper {
    private final Map<Integer, Point> _sizes;

    private static int getWidthInPixels(AdSize adSize) {
        switch (adSize.getWidth()) {
            case 0: // Interstitial.
            case -1: // Normal ads.
                return Resources.getSystem().getDisplayMetrics().widthPixels;
            default: // Deprecated ads.
                return (int) Utils.convertDpToPixel(adSize.getWidth());
        }
    }

    private static int getHeightInPixels(AdSize adSize) {
        switch (adSize.getHeight()) {
            case 0: // Interstitial.
                return Resources.getSystem().getDisplayMetrics().heightPixels;
            default: // Normal ads.
                return (int) Utils.convertDpToPixel(adSize.getHeight());
        }
    }

    @NonNull
    private static Point convertAdSizeToSize(AdSize adSize) {
        int width = getWidthInPixels(adSize);
        int height = getHeightInPixels(adSize);
        return new Point(width, height);
    }

    FacebookBannerHelper() {
        _sizes = new HashMap<>();
        for (int i = 0; i < 5; ++i) {
            AdSize adSize = getAdSize(i);
            Point size = convertAdSizeToSize(adSize);
            _sizes.put(i, size);
        }
    }

    @NonNull
    AdSize getAdSize(int index) {
        if (index == 0) {
            return AdSize.BANNER_HEIGHT_50;
        }
        if (index == 1) {
            return AdSize.BANNER_HEIGHT_90;
        }
        if (index == 2) {
            return AdSize.INTERSTITIAL;
        }
        if (index == 3) {
            return AdSize.RECTANGLE_HEIGHT_250;
        }
        if (index == 4) {
            return AdSize.BANNER_320_50;
        }
        assertThat(false).isTrue();
        return AdSize.BANNER_320_50;
    }

    @NonNull
    Point getSize(int index) {
        return Objects.requireNonNull(_sizes.get(index));
    }
}

File: proj.android_studio/ee-x-facebook-ads/src/main/java/com/ee/facebook/FacebookBannerHelper.kt
package com.ee.facebook

import android.content.res.Resources
import android.graphics.Point
import androidx.annotation.AnyThread
import com.ee.core.internal.Utils
import com.facebook.ads.AdSize
import com.google.common.truth.Truth.assertThat

internal class FacebookBannerHelper {
    companion object {
        @AnyThread
        private fun getWidthInPixels(adSize: AdSize): Int {
            return when (adSize.width) {
                0, -1 -> Resources.getSystem().displayMetrics.widthPixels
                else -> Utils.convertDpToPixel(adSize.width.toDouble()).toInt()
            }
        }

        @AnyThread
        private fun getHeightInPixels(adSize: AdSize): Int {
            return when (adSize.height) {
                0 -> Resources.getSystem().displayMetrics.heightPixels
                else -> Utils.convertDpToPixel(adSize.height.toDouble()).toInt()
            }
        }

        @AnyThread
        private fun convertAdSizeToSize(adSize: AdSize): Point {
            val width = getWidthInPixels(adSize)
            val height = getHeightInPixels(adSize)
            return Point(width, height)
        }
    }

    private val _sizes: MutableMap<Int, Point> = HashMap()

    init {
        for (i in 0..4) {
            val adSize = getAdSize(i)
            val size = convertAdSizeToSize(adSize)
            _sizes[i] = size
        }
    }

    @AnyThread
    fun getAdSize(index: Int): AdSize {
        if (index == 0) {
            return AdSize.BANNER_HEIGHT_50
        }
        if (index == 1) {
            return AdSize.BANNER_HEIGHT_90
        }
        if (index == 2) {
            return AdSize.INTERSTITIAL
        }
        if (index == 3) {
            return AdSize.RECTANGLE_HEIGHT_250
        }
        if (index == 4) {
            return AdSize.BANNER_320_50
        }
        assertThat(false).isTrue()
        return AdSize.BANNER_320_50
    }

    @AnyThread
    fun getSize(index: Int): Point {
        return _sizes[index] ?: throw IllegalArgumentException("Invalid size index")
    }
}